cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringServerHandler.java

```diff
@@ -89,16 +93,17 @@ public void init(HandlerContext context) {
   }
 
   /**
-   * Sends control commands to the client.
+   * Sends control commands to the client. Also receives program status updates from the client.
```

Comment:
I'm confused by this comment, how is this used both for sending commands to the client and also receiving updates from the client? Isn't this purely called by the client? Or is this called by both somehow?